android/app/src/main/java/com/comapeo/WifiModule.kt

```diff
@@ -0,0 +1,183 @@
+import android.os.Build
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+
+class WifiModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+    private val context = reactContext.applicationContext
+    private val connectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    private val wifiManager = context.getSystemService(Context.WIFI_SERVICE) as? WifiManager
+
+    private val networkCallback =
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            object : ConnectivityManager.NetworkCallback(FLAG_INCLUDE_LOCATION_INFO) {
+                override fun onAvailable(network: Network) {
+                    emit(network)
+                }
+
+                override fun onLost(network: Network) {
+                    emit(null)
+                }
+
+                override fun onCapabilitiesChanged(
+                    network: Network,
+                    networkCapabilities: NetworkCapabilities
+                ) {
+                    emit(network, networkCapabilities = networkCapabilities)
+                }
+
+                override fun onLinkPropertiesChanged(
+                    network: Network,
+                    linkProperties: LinkProperties
+                ) {
+                    emit(network, linkProperties = linkProperties)
+                }
+            }
+        } else {
+            object : ConnectivityManager.NetworkCallback() {
+                override fun onAvailable(network: Network) {
+                    emit(network)
+                }
+
+                override fun onLost(network: Network) {
+                    emit(null)
+                }
+
+                override fun onCapabilitiesChanged(
+                    network: Network,
+                    networkCapabilities: NetworkCapabilities
+                ) {
+                    emit(network, networkCapabilities = networkCapabilities)
+                }
+
+                override fun onLinkPropertiesChanged(
+                    network: Network,
+                    linkProperties: LinkProperties
+                ) {
+                    emit(network, linkProperties = linkProperties)
+                }
+            }
+        }
+
+    private var numberOfListeners = 0
+
+    override fun getName() = "WifiModule"
+
+    @ReactMethod
+    fun addListener(eventName: String) {
+        if (numberOfListeners == 0) {
+            startListening()
+        }
+        numberOfListeners++
+    }
+
+    @ReactMethod
+    fun removeListeners(count: Int) {
+        numberOfListeners -= count
+        if (numberOfListeners == 0) {
+            stopListening()
+        }
+    }
```

Comment:
See the [relevant React Native docs](https://reactnative.dev/docs/next/native-modules-android?android-language=kotlin#sending-events-to-javascript) for sending events from a native module.